src/components/map/Layers/Countries.tsx

```diff
@@ -1,50 +1,85 @@
 import { useState, useEffect, useContext } from "react"
 import { AppContext } from "context";
 import ReactDOMServer from "react-dom/server";
-import { EstimateGradePrevalence } from "types";
+import { EstimateGradePrevalence, CountriesMapConfig } from "types";
 import mapboxgl from "mapbox-gl";
 import CountryPopup from 'components/map/Popups/CountryPopup'
+import PartnershipsConfig from '../../../PartnershipsConfig'
+import { useHistory } from 'react-router-dom';
+
+const COUNTRY_LAYER_ID = 'Countries';
 
 // Maps estimate grade prevalence data to a match ISO3 code in the countries feature layer
 function SetCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
+
+    map.removeFeatureState({
+        source: COUNTRY_LAYER_ID,
+        sourceLayer: COUNTRY_LAYER_ID
+        });
+
     estimateGradePrevalences.forEach((country: EstimateGradePrevalence) => {
         if (country && country.testsAdministered && country.alpha3Code) {
+
+            let partnershipConfig: any = PartnershipsConfig.Country.find(x => x.iso3 === country.alpha3Code)
+
+            //PartnershipsConfig.Country.hasOwnProperty(country.alpha3Code)
+
             map.setFeatureState(
                 {
-                    source: "Countries",
-                    sourceLayer: "Countries",
+                    source: COUNTRY_LAYER_ID,
+                    sourceLayer: COUNTRY_LAYER_ID,
                     id: country.alpha3Code,
                 },
                 {
                     hasData: true,
+                    isHighlighted: false,
                     testsAdministered: country.testsAdministered,
                     geographicalName: country.geographicalName,
                     numberOfStudies: country.numberOfStudies,
                     localEstimate: country.localEstimate,
                     nationalEstimate: country.nationalEstimate,
                     regionalEstimate: country.regionalEstimate,
                     sublocalEstimate: country.sublocalEstimate,
+                    partnershipConfig: partnershipConfig
                 }
             );
         }
     });
 }
 
+function SetStudiesMode(map: mapboxgl.Map, iso3Name: string) {
```

Comment:
This function is invoked when we hover over a country right? 

I think naming is a bit misleading, let's call this `HighlightCountry` or something